packages/ra-core/src/controller/input/useReferenceArrayInputController.ts

```diff
@@ -246,37 +266,23 @@ export const useReferenceArrayInputController = (
         [queryFilter, defaultFilter, filterToQuery]
     );
 
-    const {
-        data: referenceRecordsFetched,
-        loaded,
-        refetch: refetchGetMany,
-    } = useGetMany(reference, idsToFetch || EmptyArray);
-
-    const referenceRecords = referenceRecordsFetched
-        ? referenceRecordsFetched.concat(referenceRecordsFromStore)
-        : referenceRecordsFromStore;
-
     // filter out not found references - happens when the dataProvider doesn't guarantee referential integrity
     const finalReferenceRecords = referenceRecords.filter(Boolean);
 
     const isGetMatchingEnabled = enableGetChoices
         ? enableGetChoices(finalFilter)
         : true;
+
     const {
         data: matchingReferences,
-        ids: matchingReferencesIds,
         total,
+        error: errorGetList,
+        isLoading: isLoadingGetList,
         refetch: refetchGetMatching,
-    } = useGetMatching(
+    } = useGetList(
```

Comment:
Does it mean we will remove the `getMatching` method completely?

Reply:
There is no getMatching method in dataProvider - only getManyReference.

Reply:
True. Are we removing `useGetMatching` then? And by the way, is `getManyReference` really necessary?

Reply:
useGetMatching is removed in this PR. 

getManyReference is stille useful for APIs with a hierarchical structure, e.g. `/posts/123/comments`

Reply:
Sorry, useGetMatching isn't removed yet, I'll remove it.